File: Commons/Math/src/main/java/com/greentree/common/math/vector/VectorNf.java
package com.greentree.common.math.vector;

import java.util.Arrays;

public class VectorNf extends AbstractVectorNf {
	private static final long serialVersionUID = 1L;

	private final float[] values;

	public VectorNf(VectorNf vec) {
		this.values = vec.values.clone();
	}
	
	public VectorNf(AbstractFloatVector vec) {
		this(vec.size());
		for(int i = 0; i < vec.size(); i++)
			values[i] = vec.get(i);
	}
	
	@Override
	public String toString() {
		return "VectorNf " + Arrays.toString(values);
	}

	public VectorNf(float...values) {
		this.values = values.clone();
	}
	
	public VectorNf(int length) {
		values = new float[length];
	}
	
	@Override
	public float get(int i) {
		return values[i];
	}

	@Override
	public void set(int i, float x) {
		values[i] = x;
	}

	@Override
	public int size() {
		return values.length;
	}

}

File: Commons/Math/src/main/kotlin/com/greentree/commons/math/vector/VectorNf.kt
package com.greentree.commons.math.vector

class VectorNf(private val values: FloatArray) : AbstractMutableFloatVector {

    override val size
        get() = values.size

    constructor(length: Int) : this(FloatArray(length))

    override fun get(i: Int): Float {
        return values[i]
    }

    override fun set(i: Int, x: Float) {
        values[i] = x
    }

}